### PR TITLE
Recipe app - create inventory models

### PR DESCRIPTION
**Pull request by oluyaratosin123**
- create inventory `models`
- add validation to inventory model
- generate model specs for inventories

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby '3.0.3'
+ruby '3.1.1'
```
Comment (guerreiropedr0) [inline Gemfile:4]: I think it's best to work with the same ruby version as the other collaborators of this project. The ruby version that was discussed to use was 3.0.3, so let's all kindly use it.

File: Gemfile [hunk 2]
```diff
@@ -71,3 +71,11 @@ group :development do
   # Speed up commands on slow machines / big apps [https://github.com/rails/spring]
   # gem "spring"
 end
+
+group :test do
+  gem 'capybara'
```
Comment (guerreiropedr0) [inline Gemfile:76]: Capybara is a gem we are going to need, but not yet. Let's Keep it Simple by not adding gems we aren't going to use right away. Later in the project we have a group project to setup capybara so kindly remove this gem and do: bundle install

File: app/models/inventory.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Inventory < ApplicationRecord
+  validates :name, presence: true
+
+  belongs_to :user, class_name: 'user', foreign_key: 'user_id'
+  has_many :recipes
+
+  def three_most_recent_inventories
+    inventories.order(created_at: :desc).limit(3)
+  end
```
Comment (guerreiropedr0) [inline app/models/inventory.rb:9]: This is a good method but we are not going to need it in this project, kindly remove this

File: app/models/inventory.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Inventory < ApplicationRecord
+  validates :name, presence: true
+
+  belongs_to :user, class_name: 'user', foreign_key: 'user_id'
+  has_many :recipes
```
Comment (guerreiropedr0) [inline app/models/inventory.rb:5]: The Inventory doesn't take recipes, we can safely remove this line

File: spec/models/inventory_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe Inventory, type: :model do
+  describe 'Inventory model' do
+    inventory_queries { Inventory.new(name: 'Edward', user_id: 1) }
+  end
```
Comment (guerreiropedr0) [inline spec/models/inventory_spec.rb:6]: [OPTIONAL] I think that we shouldn't use describe in this context, and preferably use the factory bot rails gem to create an inventory:
let(:inventory) { FactoryBot.create(:inventory, name: 'Edward', user_id: 1) }

**Review by guerreiropedr0 — CHANGES_REQUESTED** [review]
### **Hi @oluyaratosin123 ,** 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## **Invalid Code Review Request**

You have submitted a project, that _**Please fix the linters issue and check it please before sending the PR for review**_

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/models/inventory_spec.rb (hunk 5 above)
Comment (oluyaratosin123) [reply spec/models/inventory_spec.rb:6]: Hello @guerreiropedr0 , I will make changes on this later in other tasks, thanks for the feedback

File: spec/models/inventory_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+RSpec.describe Inventory, type: :model do
+  describe 'Inventory model' do
+    let(:inventory) { FactoryBot.create(:inventory, name: 'Edward', user_id: 1) }
+  end
+
+  before { inventory_queries.save }
+
+  it 'should check for valid attributes' do
+    expect(inventory_queries).to be_valid
```
Comment (gtekle) [inline spec/models/inventory_spec.rb:11]: - [ ] Can you help me understand where the `inventory_queries` variable is assigned with an inventory model object? How about we use the `inventory` object created using FactoryBot? what do you think?

**Review by gtekle — CHANGES_REQUESTED** [review]
Hi @oluyaratosin123 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Highlights
:heavy_check_mark: No linter errors
:heavy_check_mark: Good use of Gitflow

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly check the inline comments.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/inventory_spec.rb (hunk 6 above)
Comment (oluyaratosin123) [reply spec/models/inventory_spec.rb:11]: Hello @gtekle,
Thanks for the feedback I was suppose to use the `inventory` as the variable not `inventory_queries`

**Review by gtekle — APPROVED** [review]
Hi @oluyaratosin123 :wave:, 

# STATUS: APPROVED :clap: :clap: :clap: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
approved-stamp-02


## Highlights
:heavy_check_mark: Meaningful commit messages
:heavy_check_mark: Done all required changes

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**